tests/vlan/test_vlan.py

```diff
@@ -467,3 +479,44 @@ def test_vlan_tc5_untagged_non_broadcast(ptfadapter, vlan_ports_list, duthost):
         logger.info ("Untagged packet successfully sent from port {} to port {}".format(dst_port, src_port))
     else:
         pytest.fail("Expected packet was not received")
+
+
+def test_vlan_tc6_tagged_qinq_switch_on_outer_tag(ptfadapter, vlan_ports_list, duthost):
+    """
+    Test case #6
+    Send qinq packets w/ src and dst specified over tagged ports in vlan
+    Verify that the qinq packet is switched based on outer vlan tag + src/dst mac
+    """
+
+    # Add more supported platforms to the list as they are tested
+    qinq_switching_supported_platforms = ['mellanox']
+    if duthost.facts["asic_type"] not in qinq_switching_supported_platforms:
+        pytest.skip("Unsupported platform")
+
+    vlan_ids = vlan_ports_list[0]['permit_vlanid'].keys()
+    tagged_test_vlan = vlan_ids[0]
+
+    ports_for_test = []
+    for vlan_port in vlan_ports_list:
+        if vlan_port['pvid'] != tagged_test_vlan:
+            ports_for_test.append(vlan_port['port_index'][0])
+
+    #take two tagged ports for test
+    src_port = ports_for_test[0]
+    dst_port = ports_for_test[-1]
+
+    src_mac = ptfadapter.dataplane.get_mac(0, src_port)
+    dst_mac = ptfadapter.dataplane.get_mac(0, dst_port)
+
+    transmit_qinq_pkt = build_qinq_packet(outer_vlan_id=tagged_test_vlan, vlan_id=250, src_mac=src_mac, dst_mac=dst_mac)
+    logger.info ("QinQ packet to be sent from port {} to port {}".format(src_port, dst_port))
+    testutils.send(ptfadapter, src_port, transmit_qinq_pkt)
+
+    result_dst_if = testutils.dp_poll(ptfadapter, device_number=0, port_number=dst_port,
```

Comment:
Can we use `testutils.verify_packet` instead of `dp_poll`?

Reply:
Thanks for the comment, changed to ```testutils.verify_packet```